### PR TITLE
Ddf 2885 Add templates for eventing and sources

### PR DESCRIPTION
**Pull request by ricklarsen**
#### What does this PR do?

Adds template support for Integrating/Eventing and Integrating/Sources
#### Who is reviewing it?

@vinamartin @mcalcote @Lambeaux @jrnorth 
#### Select at least one member from relevant component team(s) from below (at least one component team member needs to approve the PR).

[Docs](https://github.com/orgs/codice/teams/docs)

#### Choose 2 committers to review/merge the PR.

@coyotesqrl
@lessarderic
@pklinef
@shaundmorris

#### How should this be tested? (List steps with links to updated documentation)

Review the jdocumentation.html/.pdf for content to be included.

#### What are the relevant tickets?
[DDF-2885](https://codice.atlassian.net/browse/DDF-2885)

#### Screenshots (if appropriate)

[jdocumentation.pdf](https://github.com/codice/ddf/files/922443/jdocumentation.pdf)

#### Checklist:
- [x] Documentation Updated
- [ ] Change Log Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

**Comment by vinamartin** [comment]
This PR looks fine to me, but I wonder if there's a more elegant solution for this: 
```
 +:federated: x
 +:connected:
 +:catalogprovider:
 +:storageprovider:
 +:catalogstore:
```

I don't have any great suggestions, but this seems a little much to me.

File: distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc [hunk 1]
```diff
@@ -0,0 +1,152 @@
+:title: Developing Sources
+:type: developingComponent
+:status: published
+:link: _developing_sources
+:summary: Creating a custom source
+
```
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc]: These files are unchanged from this point on from the ones in `src/main/resources/_contents`

File: distribution/docs/src/main/jdocs/content/_developingComponents/developing-endpoints-contents.adoc [hunk 2]
```diff
@@ -0,0 +1,59 @@
+:title: Developing Endpoints
+:type: developingComponent
+:status: published
+:link: _developing_endpoints
+:summary: Creating a custom endpoint
```
Comment (emmberk) [inline distribution/docs/src/main/jdocs/content/_developingComponents/developing-endpoints-contents.adoc]: Missing a period at the end of this summary

File: distribution/docs/src/main/jdocs/content/_developingComponents/developing-endpoints-contents.adoc [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+. Create a Java class that implements the endpoint's business logic. Example: Creating a web service that external clients can invoke.
+
+. Add the endpoint's business logic, invoking `CatalogFramework` calls as needed.  
+
+. Import the ${branding} packages to the bundle's manifest for run-time (in addition to any other required packages): +
+`Import-Package: ddf.catalog, ddf.catalog.*`
+
+. Retrieve an instance of `CatalogFramework` from the OSGi registry. (Refer to <<_working_with_osgi_-_service_registry,Working with OSGi - Service Registry>> for examples.)
+
+Deploy the packaged service to ${branding}.
```
Comment (emmberk) [inline distribution/docs/src/main/jdocs/content/_developingComponents/developing-endpoints-contents.adoc]: This step doesn't show up in the list
image

File: distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc [hunk 4]
```diff
@@ -0,0 +1,152 @@
+:title: Developing Sources
+:type: developingComponent
+:status: published
+:link: _developing_sources
+:summary: Creating a custom source
```
Comment (emmberk) [inline distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc]: Missing a period at the end of this summary

File: distribution/docs/src/main/jdocs/content/_sources/opensearch-source-contents.adoc [hunk 5]
```diff
@@ -0,0 +1,135 @@
+:title: OpenSearch Source
+:type: source
+:status: published
+:link: _opensearch_source
+:summary: Performs OpenSearch queries for metadata
```
Comment (emmberk) [inline distribution/docs/src/main/jdocs/content/_sources/opensearch-source-contents.adoc]: Missing a period at the end of this summary

File: distribution/docs/src/main/jdocs/content/_developingComponents/developing-endpoints-contents.adoc (hunk 2 above)
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_developingComponents/developing-endpoints-contents.adoc]: fixed.

File: distribution/docs/src/main/jdocs/content/_developingComponents/developing-endpoints-contents.adoc (hunk 3 above)
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_developingComponents/developing-endpoints-contents.adoc]: fixed.

File: distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc (hunk 4 above)
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc]: fixed.

File: distribution/docs/src/main/jdocs/content/_sources/opensearch-source-contents.adoc (hunk 5 above)
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_sources/opensearch-source-contents.adoc]: fixed.

File: distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+|                  |               |                                                           |
+|                  |{s}Data Store  |                                                           |
+|                  +---------------+                                                           |
++----------------------------------------------------------------------------------------------+
+....
+
+==== Implement a Source Interface
+
+There are three types of sources that can be created.
+All of these types of sources can perform a query operation.
```
Comment (mcalcote) [inline distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc]: I would combine these two sentences and remove the third (it is confusing).

"There are three types of sources that can be created, each of which can perform query operations. 
All of these sources ..."

File: distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+Operating on queries is the foundation for all sources.
+All of these sources must also be able to return their availability and the list of content types currently stored in their back-end data stores.
+
+Catalog Provider:: `ddf.catalog.source.CatalogProvider` _is used to communicate with back-end storage and allows for Query and Create/Update/Delete operations._
+Federated Source:: `ddf.catalog.source.FederatedSource` _is used to communicate with remote systems and only allows query operations._
+Connected Source:: `ddf.catalog.source.ConnectedSource` _is similar to a Federated Source with the following exceptions:_
+* _Queried on all local queries_
+* _`SiteName` is hidden (masked with the ${branding} sourceId) in query results_
+* _`SiteService` does not show this Source's information separate from ${branding}'s._
+Catalog Store:: `catalog.store.interface` is used to
```
Comment (mcalcote) [inline distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc]: Unfinished sentence?

File: distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+
+The procedure for implementing any of the source types follows a similar format:
+
+. Create a new class that implements the specified Source interface and `ConfiguredService`.
+. Implement the required methods.
```
Comment (mcalcote) [inline distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc]: Create a new class that implements the specified Source interface, the `ConfiguredService` and the required methods.

File: distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+===== Developing Connected Sources
+
+Create a custom implementation of a connected source.
+
+. Create a Java class that implements `ConnectedSource` and `ConfiguredService`. +
+`public class TestConnectedSource implements ddf.catalog.source.ConnectedSource, ddf.catalog.service.ConfiguredService`
+. Implement the required methods of the `ddf.catalog.source.ConnectedSource` and `ddf.catalog.service.ConfiguredService` interfaces.
+. Import the ${branding} interface packages to the bundle manifest (in addition to any other required packages). +
+`Import-Package: ddf.catalog, ddf.catalog.source`
+. Export the service to the OSGi registry.
```
Comment (mcalcote) [inline distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc]: Do we have a section that describes how to do this?

File: distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+
+.Connected Source Blueprint example
+[source,xml,linenums]
+----
+<service ref="[[TestConnectedSource]]" interface="ddf.catalog.source.ConnectedSource" />
+----
+[IMPORTANT]
+====
+In some Providers that are created, there is a need to make Web Service calls through JAXB clients.
+It is best NOT to create your JAXB client as a global variable.
```
Comment (mcalcote) [inline distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc]: to NOT

File: distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc (hunk 6 above)
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc]: updated.

File: distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc (hunk 7 above)
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc]: fixed.

File: distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc (hunk 8 above)
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc]: updated.

File: distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc (hunk 9 above)
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc]: I don't believe so.

File: distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc (hunk 10 above)
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc]: updated.

File: distribution/docs/src/main/jdocs/content/_sources/sources-intro-contents.adoc [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+====
+Remote sources currently extend the `ResourceReader` interface.
+However, a `RemoteSource` is not treated as a `ResourceReader`.
+The `getSupportedSchemes()` method should never be called on a `RemoteSource`, thus the suggested implementation for a `RemoteSource` is to return an empty set.
+The `retrieveResource( ... )` and `getOptions( ... )` methods will be called and MUST be properly implemented by a `RemoteSource`.
+====
+
+[[_federated_source]]
+<<_federated_sources,Federated Sources>>:: A federated source is a remote source that can be included in federated queries by request or as part of an enterprise query. Federated sources support query and site information operations only. Catalog modification operations, such as create, update, and delete, are not allowed.
+Federated sources also expose an event service, which allows the Catalog Framework to subscribe to even notifications when metacards are created, updated, and deleted.
```
Comment (coyotesqrl) [inline distribution/docs/src/main/jdocs/content/_sources/sources-intro-contents.adoc]: ❗️ s/b "...subscribe to event..."

File: distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc [hunk 12]
```diff
@@ -0,0 +1,149 @@
...
+[source,xml,linenums]
+----
+<service ref="[[TestConnectedSource]]" interface="ddf.catalog.source.ConnectedSource" />
+----
+[IMPORTANT]
+====
+In some Providers that are created, there is a need to make Web Service calls through JAXB clients.
+It is best to NOT create your JAXB client as a global variable.
+There may be intermittent failures with the creation of Providers and federated sources when clients are created in this manner.
+Create your JAXB clients every single time within the methods that require it in order to avoid this issue.
```
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc]: fixing here and in the original content.

Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc]: also removed the `your JAXB client`

File: distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc [hunk 13]
```diff
@@ -0,0 +1,149 @@
...
+===== Exception Handling
+
+In general, sources should only send information back related to the call, not implementation details.
+
+====== Exception Examples
+
+Follow these guidelines for effective exception handling:
+
+* Use a "Site XYZ not found" message rather than the full stack trace with the original site not found exception.
+* IF the caller issues a malformed search request, return an error describing the right form, or specifically what was not recognized in the request. Do not return the exception and stack trace where the parsing broke.
```
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc]: fixing in both copies.

File: distribution/docs/src/main/jdocs/content/_eventing/subscriptions-contents.adoc [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+*Subscriptions are not persisted by the Catalog itself.* +
+Subscriptions must be explicitly persisted by an endpoint and are not persisted by the Catalog.
+The Catalog Framework, or more specifically the Event Processor itself, does not persist subscriptions.
+Certain endpoints, however, can persist the subscriptions on their own and recreate them on system startup.
+====
+
+==== Creating a Subscription
+
+Currently, the Catalog reference implementation does not contain a subscription endpoint.
+Nevertheless, an endpoint that exposes a web service interface to create, update, and delete subscriptions would provide a client's subscription's filtering criteria to be used by Catalog's Event Processor to determine which create, update, and delete events are of interest to the client.
```
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_eventing/subscriptions-contents.adoc]: I'm not 100% sure, as that is a very convoluted sentence. Attempting to simplify.

File: distribution/docs/src/main/jdocs/content/_sources/confluence-federated-source-contents.adoc [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+
+Confluence Source has been tested against the following versions of Confluence with REST API v2
+
+* Confluence 1000.444.5 (Cloud)
+* Confluence 5.10.6 (Server)
+* Confluence 5.10.7 (Server)
+
+===== Installing the Confluence Federated Source
+
+The CSW Federated Source is installed by default with a standard installation in the ${ddf-spatial} application.
```
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_sources/confluence-federated-source-contents.adoc]: fixed.

File: distribution/docs/src/main/jdocs/content/_sources/confluence-federated-source-contents.adoc [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+
+Add a New CSW Federated Source through the ${admin-console}:
```
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_sources/confluence-federated-source-contents.adoc]: fixed.

File: distribution/docs/src/main/jdocs/content/_sources/confluence-federated-source-contents.adoc [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+. Navigate to the *${admin-console}*.
+. Select the *${ddf-catalog}* application.
+. Select the *Sources* tab.
+. Add a New source.
+. Name the New source.
+. Select *Confluence Federated Source* from *Binding Configurations*.
+
+===== Configuring the Confluence Federated Source
+
+Configure an Existing CSW Federated Source through the ${admin-console}:
```
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_sources/confluence-federated-source-contents.adoc]: fixed.

File: distribution/docs/src/main/jdocs/content/_sources/confluence-federated-source-contents.adoc [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+. Navigate to the *${admin-console}*.
+. Select the *${ddf-catalog}* application.
+. Select the *Sources* tab.
+. Select the name of the source to edit.
+
+include::{adoc-include}/_tables/Confluence_Federated_Source-table-contents.adoc[]
+
+===== Usage Limitations of the Confluence Federated Source
+
+Most of the fields that can be queried on confluence have some sort of restriction on them. Most of the fields do not support the `like` aka `~` operation so the source will convert `like` queries to `equal` queries for attributes that don't support `like`. If the source receives a query with attributes it doesn't understand, it will just ignore them. If the query doesn't contain any attributes that map to confluence search attributes, an empty result set will be returned.
```
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_sources/confluence-federated-source-contents.adoc]: fixed.

File: distribution/docs/src/main/jdocs/content/_sources/confluence-federated-source-contents.adoc [hunk 19]
```diff
@@ -0,0 +1,53 @@
...
+
+Depending on your version of Confluence, when downloading attachments you might get redirected to a different download URL. If this happens the default URLResourceReader configuration does not allow redirects and will fail the download. This can be fixed by enabling redirects in the URLResourceReader configuration.
```
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_sources/confluence-federated-source-contents.adoc]: done and done.

File: distribution/docs/src/main/jdocs/content/_sources/csw-connected-source-contents.adoc [hunk 20]
```diff
@@ -0,0 +1,45 @@
...
+:status: published
+:link: _csw_connected_source
+:summary: Searches collections of descriptive information (metadata) for data, services, and related information objects.
+:federated:
+:connected: x
+:catalogprovider:
+:storageprovider:
+:catalogstore:
+
+The CSW connected source supports the ability to search collections of descriptive information (metadata) for data, services, and related information objects.
```
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_sources/csw-connected-source-contents.adoc]: fixed.

File: distribution/docs/src/main/jdocs/content/_sources/csw-connected-source-contents.adoc [hunk 21]
```diff
@@ -0,0 +1,45 @@
...
+
+* Navigate to the *${admin-console}*.
+* Select the *${ddf-catalog}* application.
+* Add a New source.
+* Name the New source.
+* Select *CSW Connected Source* from *Binding Configurations*.
+
+===== Configuring the CSW Connected Source
+
+Configure an Existing CSW Federated Source through the ${admin-console}:
```
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_sources/csw-connected-source-contents.adoc]: fixed.

File: distribution/docs/src/main/jdocs/content/_sources/csw-federated-source-contents.adoc [hunk 22]
```diff
@@ -0,0 +1,44 @@
...
+The CSW Federated Source is installed by default with a standard installation in the ${ddf-spatial} application.
+
+Add a New CSW Federated Source through the ${admin-console}:
+
+* Navigate to the *${admin-console}*.
+* Select the *${ddf-catalog}* application.
+* Select the *Sources* tab.
+* Add a New source.
+* Name the New source.
+* Select *CSW Federated Source* from *Binding Configurations*.
```
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_sources/csw-federated-source-contents.adoc]: I can't disagree with the naming. I _think_ I have addressed your comment.

File: distribution/docs/src/main/jdocs/content/_sources/csw-federation-profile-source-contents.adoc [hunk 23]
```diff
@@ -0,0 +1,43 @@
...
+:link: _csw_federation_profile_source
+:summary: Queries a CSW version 2.0.2 compliant service.
+:federated: x
+:connected:
+:catalogprovider:
+:storageprovider:
+:catalogstore:
+
+
+The CSW federation profile source supports the ability to search collections of descriptive information (metadata) for data, services, and related information objects.
```
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_sources/csw-federation-profile-source-contents.adoc]: fixed.

File: distribution/docs/src/main/jdocs/content/_sources/federation-strategy-contents.adoc [hunk 24]
```diff
@@ -0,0 +1,122 @@
...
+|                | Catalog Provider  |                       |
+|                +-------------------+                       |
+|                |cDEF               |                       |
+|                | Storage Provider  |                       |
+|                \-------------------/                       |
++------------------------------------------------------------+
+....
+
+The Catalog normalizes incoming queries into an OGC Filter format.
+When the query is disseminated by the Catalog Framework to the sources, each source is responsible for denormalizing the OGC Filter formatted query into the format understood by the external store that the source is acting as a proxy.
```
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_sources/federation-strategy-contents.adoc]: updated.

File: distribution/docs/src/main/jdocs/content/_sources/sources-intro-contents.adoc (hunk 11 above)
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_sources/sources-intro-contents.adoc]: fixed.

**Pull request merged**